dockertest/output.py

```diff
@@ -798,29 +798,32 @@ def mustfail(cmdresult, expected_status=None, failmsg=None):
     """
     # FIXME: temporary: backward compatibility for pre-20160330 code
     # FIXME: remove before the next API-changing release
-    if expected_status is None:
-        expected_status = 1
-    if isinstance(expected_status, basestring):
-        failmsg = expected_status
+    if expected_status is None:                    # old: mustfail(x)
         expected_status = 1
+    if isinstance(expected_status, basestring):    # old: mustfail(x, "msg")
+        if not expected_status.isdigit():     # pylint: disable=E1101
+            failmsg = expected_status
+            expected_status = 1
 
     if failmsg is None:
         details = "%s" % cmdresult
     else:
         details = "%s\n%s" % (failmsg, cmdresult)
-    if cmdresult is not None:
```

Comment:
There was a reason for adding this, but my memory of it is long since gone.  I'm fine with killing it, and if the other thing is still a problem, it'll re-present itself :smiley: